annotator-core/src/main/java/edu/ucr/cs/riple/core/registries/field/ClassFieldRecord.java

```diff
@@ -96,6 +96,16 @@ public void addFieldDeclaration(FieldDeclaration fieldDeclaration) {
     this.fields.add(new FieldDeclarationRecord(fieldDeclaration));
   }
 
+  /**
+   * Checks if the class contains a field with the given name.
```

Comment:
This method is doing something different than the docs.  It seems to look for a single field declaration that declares _all_ the names in the `Set` given as a parameter.  We should update the docs and consider renaming the method.

Reply:
@msridhar Yes, I agree. Regarding field declarations (or `OnFIeld` instances) we have to document somewhere how they are created. Throughout the entire `core` module, every `OnField` instance contains all inline field names for the declaration, for instance for a declaration `Foo a, b, c`, all `OnField` instances targeting this declaration, contain all `a`, `b` and `c` on their `variables` property. This has been done at deserializing fixes [here](https://github.com/ucr-riple/NullAwayAnnotator/blob/f773208495a0e7a8ee2432a8f3b0da4ddb0da7a1/annotator-core/src/main/java/edu/ucr/cs/riple/core/checkers/nullaway/NullAway.java#L128). That is the reason why I used `equals` instead of checking if it contains it. Will change the name and the documentation now.

Reply:
Updated javadoc for this method 71b901df47ddf678fb4cfb7097e47a88be63b456.
Will work on a more detailed documentation with revisiting all methods working on fields in a followup PR.